src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+                    );
+                    this.setSuggestion(this.filtered_list, false, this.listkey);
+                } else {
+                    this.show_empty = true;
+                }
+            }
+            const input = new CustomEvent("input", {
+                bubbles: true , composed:true
+            });
+            this.dispatchEvent(input);
+        } else {
+            this.filtered_list = [];
+            this.show_empty = true;
+            this.setSuggestion(this.filtered_list, false, this.listkey);
+        }
+        return;
+    }
+    private _setActive(resultLength: number) {
+        if (this.prevIndex >= 0 && this.prevIndex < resultLength) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
+
+            if (selectedItem) {
+                this.private_items[this.prevIndex]?.classList.remove(
+                    "sgn-active"
+                );
+                selectedItem.classList.add("sgn-active");
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add("sgn-active");
+            this.private_items[this.prevIndex]?.classList.remove("sgn-active");
+        }
+
+        this.prevIndex = this.index;
+    }
+
+    private _setInputValue() {
+        const selectedValue = this.result[this.index][this.listkey] || this.result[this.index].name;
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+            this.show_container = false;
+        }
+    }
+    _highlight(name: any, input: string) {
+        if (input && name.toLowerCase().includes(input)) {
+            const parts = name.split(new RegExp(`(${input})`, "gi"));
+            return parts
+                .map((part: any, index: number) => {
+                    if (index > 1) {
+                        return `<span class="sgn-highlight">${part}</span>`;
+                    } else {
+                        return part;
+                    }
+                })
+                .join("");
+        }
+        return name;
+    }
+    _startsWithHighlight(name: any) {
+        const input_val = (this.input_el as HTMLInputElement)?.value;
+        if (!input_val) {
+            return name;
+        }
+
+        const escaped_input_val = input_val.replace(
+            /[-\/\\^$*+?.()|[\]{}]/g,
+            "\\$&"
+        );
+        const re = new RegExp(
+            "(" + escaped_input_val.split(" ").join("|") + ")",
+            "gi"
+        );
+        return name.replace(re, `<span class="sgn-highlight">$1</span>`);
+    }
+    setSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more: boolean,
+        key: string
```

Comment:
The specification is updated to have just two parameters. This third parameter is not needed. I see that the thir parameter is passed to the format().
Same applies to appendSuggestion.